### ISSUE TITLE
[BUG] JSON output for Validate should produce valid, parseable JSON when written to a file

### ISSUE THREAD

**Issue by SamPlatek** [aws-cloudformation/cloudformation-guard#180]
**Describe the bug**
When I run  `cfn-guard validate -o json --show-summary none -d ./my_template.yaml -r ./guard_rules/ > cfn_guard_results.json`, the resulting file should be valid JSON. However, it seems to put 1 JSON object per rule file in the directory `guard_rules/`. 

**To Reproduce**
1. Setup folder `guard_rules` and put more than 1 rule file in there with at least 1 rule in each file
2.  Setup a valid yaml template called my_template.yaml
3. Run the command `cfn-guard validate -o json --show-summary none -d ./my_template.yaml -r ./guard_rules/ > cfn_guard_results.json`

The output is structured like so:
```
{<FILE_INFO_1>}
{<FILE_INFO_2>}
{<FILE_INFO_3>}
.....
```

**NOTE: Please be sure that the templates, rules and logs you provide as part of your bug report do not contain any sensitive information.**

**Expected behavior**
The output should instead be able to be parsed as valid json.

A few ideas:

No. 1:

```
{
  <filename_1>: {FILE_INFO_1},
  <filename_2>: {FILE_INFO_2}f
}
```

No. 2:

```
[
 {FILE_INFO_1},
 {FILE_INFO_2}
]
```

**Screenshots**

**Operating System:**
agnostic

**OS Version**
n/a

**Additional context**
Additionally, while less of an issue, the yaml output puts a file break between each file's info - while this is more manageable than invalid JSON, it would be nice to have it also print out a single valid yaml document with the results

**Comment by 0xjjoyy** [comment]
As (junit)xml is not yet supported, having the complete JSON per cfn template would help here. It would then be easier to transform the JSON to junitxml to include for codebuild reports.

https://github.com/aws-cloudformation/cloudformation-guard/issues/49#issuecomment-829216602

**Comment by joshfried-aws** [comment]
Hi @SamPlatek and @0xjjoyy I am going to go ahead and close this issue since this has been addressed as part of #339. 

Feel free to reopen if need be.
